### DISCUSSION TITLE
About the lag after added outline to multiple objects

### DISCUSSION THREAD

**Discussion by KayHulk** [Arvtesh/UnityFx.Outline#48] [Q&A] [answered]
Hello! 

I must say that this Plugin is very useful, thanks very much. But I found that when I had too many GameObjects to outline in my scene, the game became very slow.

Unity version：Unity 2020.3.4

That's the case, I plan to make a box selection function to add Outline to all selected objects. So I tried it and generated 100 cubes in a new scene, and used a for loop to add an outline to each cube in the scene. After Unity was running, it gets pretty sluggish. But sometimes I have to select more than 100 objects.

Here is my code：
```
var layer = MainCamera.GetComponent<OutlineEffect>().OutlineLayers[0];

for (int i = 0; i < ObjectsParent.childCount; i++)
{
    layer.Add(ObjectsParent.GetChild(i).gameObject);
}
```

I don’t know if I did something wrong? Maybe there is any solution?

**Comment by Arvtesh** [comment, marked as answer]
This particular library is *post-processing* outlines. This means each outlined object adds one full-screen pass, which is expensive performance-wise esp. on mobiles. Post-processing outlines in general trade performance for visual quality. If you need to have many objects outlined in realtime, you'd better use different library, which does not rely on post-processing.

**Reply by KayHulk** [reply]
Oh~ Thank you for your answer! It seems I need to think of a solution, thank you very much!
